data/storeDEPRECATED/mongo/mongo.go

```diff
@@ -426,6 +428,10 @@ func (d *DataSession) CreateDataSetData(ctx context.Context, dataSet *upload.Upl
 			if err != nil {
 				return errors.Wrap(err, "unable to create cbg bucket")
 			}
+			err = d.BucketStore.UpsertMetaData(ctx, dataSet.UserID, incomingUserMetadata)
+			if err != nil {
+				return errors.Wrap(err, "unable to update metadata")
+			}
```

Comment:
We are updating metadata only when we have samples (i.e. as far as I understand only for cbg). 
I thing this upsert should be done in any case (with or without samples, maybe only when incomingUserMetadata is not nil).